.github/workflows/phpcs.yml

```diff
@@ -31,59 +35,134 @@ jobs:
-      - uses: actions/cache@v2
-        id: composer-cache
+      - name: Cache Composer dependencies
+        uses: actions/cache@v2
         with:
           path: ${{ steps.get-composer-cache-dir.outputs.dir }}
           key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
           restore-keys: |
             ${{ runner.os }}-composer-
 
-      - uses: "shivammathur/setup-php@v2"
-        with:
-          php-version: ${{ inputs.php_version }}
-      - uses: "ramsey/composer-install@v2"
-
-      - name: "Give permissions"
+      # ------------------------------------------------------------------------------
+      # Check for PHPCS
+      # ------------------------------------------------------------------------------
+      - name: Check PHPCS installation
         run: |
-          sudo chown -R root:root $GITHUB_WORKSPACE
+          if [ ! -f vendor/bin/phpcs ]; then
+            echo "PHPCS not found in vendor/bin, installing..."
+            composer require "squizlabs/php_codesniffer=*"
+          else
+            echo "PHPCS found in vendor/bin"
+          fi
+          vendor/bin/phpcs --version
 
       # ------------------------------------------------------------------------------
       # Get changed files
       # ------------------------------------------------------------------------------
       - name: Get list of changed files
-        id: files
+        id: get-changed-files
         run: |
-          echo "CHANGED_FILES=$(git diff --name-only --diff-filter=AM ${{ github.event.pull_request.base.sha }} ${{ github.event.pull_request.head.sha }} -- . ':!tests' | grep '\.php$' | tr '\n' ' ')" >> $GITHUB_ENV
+          CHANGED_FILES=$(git diff --name-only --diff-filter=AM ${{ github.event.pull_request.base.sha }} ${{ github.event.pull_request.head.sha }} -- . ':!tests' | grep '\.php$')
+          echo "CHANGED_FILES<<EOF" >> $GITHUB_ENV
+          echo "${CHANGED_FILES}" >> $GITHUB_ENV
+          echo "EOF" >> $GITHUB_ENV
+          echo "Changed files (debug): ${CHANGED_FILES}"
 
       # ------------------------------------------------------------------------------
-      # PHPCS
+      # Run PHPCS and Output Errors
       # ------------------------------------------------------------------------------
       - uses: reviewdog/action-setup@v1
         with:
           reviewdog_version: latest # Optional. [latest,nightly,v.X.Y.Z]
-      - name: Run reviewdog
+      - name: Run PHPCS and Output Errors
         env:
           REVIEWDOG_GITHUB_API_TOKEN: ${{ secrets.access-token }}
         run: |
-          # Run phpcs and capture both the output and the exit code
-          JSON_REPORT=$(vendor/bin/phpcs --report=json ${{ env.CHANGED_FILES }} || echo "")
-          PHPCS_EXIT_CODE=$?
-
-          # Check if phpcs produced a JSON report
-          if [ -z "$JSON_REPORT" ]; then
-            echo "No JSON report generated by phpcs"
-            exit $PHPCS_EXIT_CODE
+          DEBUG=true
```

Comment:
Seems like `DEBUG` is always true. Maybe take this from the action's input and have it `false` as default ?